api/v1beta1/azureclusteridentity_webhook.go

```diff
@@ -26,3 +32,30 @@ func (c *AzureClusterIdentity) SetupWebhookWithManager(mgr ctrl.Manager) error {
 		For(c).
 		Complete()
 }
+
+// +kubebuilder:webhook:verbs=create;update,path=/validate-infrastructure-cluster-x-k8s-io-v1beta1-azureclusteridentity,mutating=false,failurePolicy=fail,matchPolicy=Equivalent,groups=infrastructure.cluster.x-k8s.io,resources=azureclusteridentities,versions=v1beta1,name=validation.azureclusteridentity.infrastructure.cluster.x-k8s.io,sideEffects=None,admissionReviewVersions=v1;v1beta1
+
+var _ webhook.Validator = &AzureClusterIdentity{}
+
+// ValidateCreate implements webhook.Validator so a webhook will be registered for the type.
+func (c *AzureClusterIdentity) ValidateCreate() error {
+	var allErrs field.ErrorList
+	if c.Spec.Type == UserAssignedMSI && c.Spec.ResourceID == "" {
```

Comment:
should we also validate the inverse, ie. that resourceID is empty when the Type is not UserAssigned?

Reply:
Yes that would make sense! I didn't realize resourceID was unique to UserAssignedMSI

Reply:
Would the error be a field.Invalid or something more specific? I was thinking field.NotSupported may fit, but didn't find anything specific to an extra field.

Reply:
I think `Field.Forbidden`  might be most appropriate for that one:

``` 
// Forbidden returns a *Error indicating "forbidden".  This is used to
// report valid (as per formatting rules) values which would be accepted under
// some conditions, but which are not permitted by current conditions (e.g.
// security policy).
```